pkg/operator/ceph/object/user/controller.go

```diff
@@ -281,13 +281,59 @@ func (r *ReconcileObjectStoreUser) createorUpdateCephUser(u *cephv1.CephObjectSt
 		} else {
 			return errors.Wrapf(err, "failed to get details from ceph object user %q", u.Name)
 		}
+	} else if *user.MaxBuckets != *r.userConfig.MaxBuckets {
+		// TODO handle update for user capabilities
+		user, err = r.objContext.AdminOpsClient.ModifyUser(context.TODO(), *r.userConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create ceph object user %v", &r.userConfig.ID)
+		}
+		logCreateOrUpdate = fmt.Sprintf("updated ceph object user %q", u.Name)
+	}
+
+	var quotaEnabled = false
+	var maxSize int64 = -1
+	var maxObjects int64 = -1
+	if u.Spec.Quotas != nil {
+		if u.Spec.Quotas.MaxObjects != nil || u.Spec.Quotas.MaxSize != nil {
+			if (u.Spec.Quotas.MaxObjects != nil) && (*u.Spec.Quotas.MaxObjects != *user.UserQuota.MaxObjects) {
+				maxObjects = *u.Spec.Quotas.MaxObjects
+				quotaEnabled = true
+			}
+			if (u.Spec.Quotas.MaxSize != nil) && (u.Spec.Quotas.MaxSize.Value() != *user.UserQuota.MaxSize) {
+				maxSize = u.Spec.Quotas.MaxSize.Value()
+				quotaEnabled = true
+			}
+			if quotaEnabled {
+				userQuota := admin.QuotaSpec{
+					UID:        u.Name,
+					Enabled:    &quotaEnabled,
+					MaxSize:    &maxSize,
+					MaxObjects: &maxObjects,
+				}
+				err = r.objContext.AdminOpsClient.SetUserQuota(context.TODO(), userQuota)
+				if err != nil {
+					return errors.Wrapf(err, "failed to set quota for the user %v", u.Name)
+				}
+			}
+		} else if *user.UserQuota.Enabled {
```

Comment:
Does this allow the quota to be disabled? It seems that this is checking if the quota **is** enabled, not that the quota **is not** enabled.

Reply:
Yes `*user.UserQuota.Enabled` gives an idea about the existing state of quota at the backend. This is else of the first check `if u.Spec.Quotas != nil` .This means current value is remove or unset from the user. So I will reset the quota if it enabled earlier. I have added this part only specific for handling the reset of the quota from user end